hooks/useBattleLogic.js

```diff
@@ -96,6 +85,24 @@ export function useBattleLogic(
 		}, 1000);
 	};
 
+	const handlePlayerDamage = (damage) => {
+		setPlayerHealth((prevHealth) => prevHealth - damage);
+	};
+
+	useEffect(() => {
```

Comment:
I am not sure if you need a useEffect here. I can not give you a better solution off the cuff, but this might help you:

[you might not need an effect](https://react.dev/learn/you-might-not-need-an-effect)